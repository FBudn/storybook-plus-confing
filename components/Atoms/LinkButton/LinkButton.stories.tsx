import React from "react";
import { View } from "react-native";
import LinkButton from "./LinkButton";

const LinkButtonMeta = {
  title: "Atoms/LinkButton",
  component: LinkButton,
  argTypes: {
    onClickLink: { action: "pressed the linkButton" },
  },
  args: {
    children: "LinkButton",
    testId: "test-linkButton-id",
  },
  decorators: [
    (Story: any) => (
      <View
        style={{
          alignItems: "center",
          justifyContent: "center",
          flex: 1,
          backgroundColor: "white",
        }}
      >
        <Story />
      </View>
    ),
  ],
};

export default LinkButtonMeta;

export const Basic = {};
