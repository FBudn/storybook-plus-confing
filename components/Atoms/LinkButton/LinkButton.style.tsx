import { StyleSheet } from "react-native";

const LinkButtonStyles = StyleSheet.create({
  container: {
    justifyContent: "flex-end",
    alignItems: "flex-start",
    alignContent: "flex-start",
    textAlign: "left",
  },
  text: {
    color: "rgb(82 82 91)",
    fontWeight: "500",
    textDecorationLine: "underline",
    textDecorationColor: "rgb(82 82 91)",
    textDecorationStyle: "solid",
    fontSize: 20,
  },
});

export default LinkButtonStyles;
