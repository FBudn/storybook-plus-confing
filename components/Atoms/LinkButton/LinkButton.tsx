import React from "react";
import { Text, StyleSheet, TouchableOpacity, View } from "react-native";

export interface LinkButtonProps {
  children: React.ReactNode;
  onClickLink: () => void;
  testId: string;
}

const LinkButton: React.FC<LinkButtonProps> = ({
  children,
  onClickLink,
  testId,
}) => {
  return (
    <View>
      <TouchableOpacity
        style={styles.container}
        testID={testId}
        onPress={onClickLink}
      >
        <Text style={styles.text}> {children} </Text>
      </TouchableOpacity>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    justifyContent: "flex-end",
    alignItems: "center",
    alignContent: "center",
    textAlign: "center",
  },
  text: {
    color: "rgb(82 82 91)",
    fontWeight: "500",
    textDecorationLine: "underline",
    textDecorationColor: "rgb(82 82 91)",
    textDecorationStyle: "solid",
    fontSize: 20,
  },
});

export default LinkButton;
