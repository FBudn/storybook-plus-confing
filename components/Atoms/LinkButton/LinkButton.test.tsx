import React from "react";
import { fireEvent, render, screen } from "@testing-library/react-native";
import TestRenderer from "react-test-renderer";
import LinkButton, { LinkButtonProps } from "./LinkButton";

const mockonClickLink = jest.fn();

const props: LinkButtonProps = {
  testId: `test-linkButton-id`,
  children: `LinkButton Test`,
  onClickLink: mockonClickLink,
};

test(`Should renred LinkButton component`, () => {
  render(<LinkButton {...props} />);
  const LinkButtonElement = screen.getByTestId(`test-linkButton-id`);
  expect(LinkButtonElement).toBeOnTheScreen();
});

test(`Schould match snapshot`, () => {
  const LinkButtonElement = TestRenderer.create(
    <LinkButton {...props} />
  ).toJSON();
  expect(LinkButtonElement).toMatchSnapshot();
});

test(`Should render and have styles:`, () => {
  render(<LinkButton {...props} />);
  const LinkButtonElement = screen.getByTestId(`test-linkButton-id`);
  expect(LinkButtonElement).toBeOnTheScreen();

  expect(LinkButtonElement).toHaveStyle({
    justifyContent: "flex-end",
    alignItems: "center",
    alignContent: "center",
    textAlign: "center",
  });
});

test(`Should handleClick`, () => {
  render(<LinkButton {...props} />);
  const LinkButtonElement = screen.getByTestId(`test-linkButton-id`);
  fireEvent.press(LinkButtonElement);
  expect(mockonClickLink).toBeCalledTimes(1);
});
