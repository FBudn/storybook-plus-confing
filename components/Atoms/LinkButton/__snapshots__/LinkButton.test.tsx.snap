// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Schould match snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignContent": "center",
      "alignItems": "center",
      "justifyContent": "flex-end",
      "opacity": 1,
      "textAlign": "center",
    }
  }
  testID="test-linkButton-id"
>
  <Text
    style={
      {
        "color": "rgb(82 82 91)",
        "fontSize": 20,
        "fontWeight": "500",
        "textDecorationColor": "rgb(82 82 91)",
        "textDecorationLine": "underline",
        "textDecorationStyle": "solid",
      }
    }
  >
     
    LinkButton Test
     
  </Text>
</View>
`;
